src/main/java/filter/ParseException.java

```diff
@@ -1,8 +1,11 @@
 package filter;
 
+/**
+ *  Thrown to indicate failure to interpret a filter expression
```

Comment:
"Failure to parse" might be a bit more precise. This exception indicates a problem with syntax.